src/standalone/api/kernels/index.ts

```diff
@@ -29,18 +31,6 @@ export function getKernelsApi(extensionId: string): Kernels {
                 });
                 return;
             }
-            const execution = kernelProvider.getKernelExecution(kernel);
-            if (!isRemoteConnection(kernel.kernelConnectionMetadata) && execution.executionCount === 0) {
-                // For local kernels, execution count must be greater than 0,
-                // As we pre-warms kernels (i.e. we start kernels even though the user may not have executed any code).
-                // The only way to determine whether users executed code is to look at the execution count
```

Comment:
We need some check to ensure we do not trigger these events for kernels that are automatically started
Events should be triggered only for kernels that were explicitly started by user

Reply:
Let me know if the latest iteration based on feedback below addresses this. Thanks!